code/processes/gateway.q

```diff
@@ -398,73 +402,79 @@ getserveridstype:{[att;typ]
   attype:`cross;
 
   servers:$[typ=`all;
-		exec serverid!attributes from .gw.servers where active;
-		exec serverid!attributes from .gw.servers where active, servertype=typ];
+    exec serverid!attributes from .gw.servers where active;
+    exec serverid!attributes from .gw.servers where active,servertype=typ];
 
   if[`besteffort in key att;
-	if[-1h=type att`besteffort; besteffort:att`besteffort];
-	att:delete besteffort from att;
+  if[-1h=type att`besteffort;besteffort:att`besteffort];
+  att:delete besteffort from att;
   ];
   if[`attributetype in key att;
-	if[-11h=type att`attributetype; attype:att`attributetype];
-	att:delete attributetype from att;
+  if[-11h=type att`attributetype;attype:att`attributetype];
+  att:delete attributetype from att;
   ];
 
   res:$[attype=`independent; 
-	getserversindependent[att;servers;besteffort];
-	getserverscross[att;servers;besteffort]];
+  getserversindependent[att;servers;besteffort];
+  getserverscross[att;servers;besteffort]];
 
   serverids:first value flip $[99h=type res; key res; res];
   if[all 0=count each serverids;'"no servers match ",string[typ]," requested attributes"];
   :serverids;
  }
 
 // execute an asynchronous query
-asyncexecjpt:{[query;servertype;joinfunction;postback;timeout]
+asyncexecjpts:{[query;servertype;joinfunction;postback;timeout;sync]
  if[.gw.permissioned;
-  if[not .pm.allowed[.z.u; query];
-   @[neg .z.w;.gw.formatresponse[0b;`async;"User is not permissioned to run this query from the gateway"];()];
+  if[not .pm.allowed[.z.u;query];
+   @[neg .z.w;.gw.formatresponse[0b;sync;"User is not permissioned to run this query from the gateway"];()];
    :();
    ];
   ];
  query:({[u;q]$[`.pm.execas ~ key `.pm.execas;value (`.pm.execas; q; u);value q]}; .z.u; query);
  /- if sync calls are allowed disable async calls to avoid query conflicts
- $[.gw.synccallsallowed;errStr:.gw.errorprefix,"only synchronous calls are allowed";
+ $[.gw.synccallsallowed and .z.K<3.6;errStr:.gw.errorprefix,"only synchronous calls are allowed";
  [errStr:"";
  if[99h<>type servertype;
-	// its a list of servertypes e.g. `rdb`hdb
-	servertype,:();
-	missing:(servertype,()) except exec distinct servertype from .gw.servers where active;
-	if[count missing; errStr:.gw.errorprefix,"not all of the requested server types are available; missing "," " sv string missing];
-	queryattributes:()!();
+  // its a list of servertypes e.g. `rdb`hdb
+  servertype:distinct servertype,();
+  missing:(servertype,()) except exec distinct servertype from .gw.servers where active;
```

Comment:
Line 440 will ensure `servertype` is a list, can trim this down to `servertype except ...`. (+ extra space indentation for clarity).